### ISSUE TITLE
Segfault: Invalid read of size 8 llvm::BasicBlock::getTerminator()

### ISSUE THREAD

**Issue by Jack-Clark** [microsoft/DirectXShaderCompiler#4450]
Running `./build/bin/dxc -T cs_6_7 segfault.hlsl` from the [attached archive](https://github.com/microsoft/DirectXShaderCompiler/files/8665745/segfault.tar.gz) causes a segfault on Ubuntu 20.04. When I run it with `valgrind -v ./build/bin/dxc -T cs_6_7 segfault.hlsl` I get the following:
```
==93131== Memcheck, a memory error detector
==93131== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==93131== Using Valgrind-3.15.0-608cb11914-20190413 and LibVEX; rerun with -h for copyright info
==93131== Command: ./build/bin/dxc -T cs_6_7 /home/jack/code/mc-imperial/spirv-control-flow/fleshing/bugs/dxc/segfault/segfault.hlsl
==93131== 
--93131-- Valgrind options:
--93131--    -v
--93131-- Contents of /proc/version:
--93131--   Linux version 5.13.0-40-generic (buildd@ubuntu) (gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022
--93131-- 
--93131-- Arch and hwcaps: AMD64, LittleEndian, amd64-cx16-lzcnt-rdtscp-sse3-ssse3-avx-avx2-bmi-f16c-rdrand
--93131-- Page sizes: currently 4096, max supported 4096
--93131-- Valgrind library directory: /usr/lib/x86_64-linux-gnu/valgrind
--93131-- Reading syms from /home/jack/code/DirectXShaderCompiler/build/bin/dxc-3.7
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/ld-2.31.so
--93131--   Considering /usr/lib/x86_64-linux-gnu/ld-2.31.so ..
--93131--   .. CRC mismatch (computed 0306b78c wanted 8d362b37)
--93131--   Considering /lib/x86_64-linux-gnu/ld-2.31.so ..
--93131--   .. CRC mismatch (computed 0306b78c wanted 8d362b37)
--93131--   Considering /usr/lib/debug/lib/x86_64-linux-gnu/ld-2.31.so ..
--93131--   .. CRC is valid
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/valgrind/memcheck-amd64-linux
--93131--    object doesn't have a symbol table
--93131--    object doesn't have a dynamic symbol table
--93131-- Scheduler: using generic scheduler lock implementation.
--93131-- Reading suppressions file: /usr/lib/x86_64-linux-gnu/valgrind/default.supp
==93131== embedded gdbserver: reading from /tmp/vgdb-pipe-from-vgdb-to-93131-by-jack-on-???
==93131== embedded gdbserver: writing to   /tmp/vgdb-pipe-to-vgdb-from-93131-by-jack-on-???
==93131== embedded gdbserver: shared mem   /tmp/vgdb-pipe-shared-mem-vgdb-93131-by-jack-on-???
==93131== 
==93131== TO CONTROL THIS PROCESS USING vgdb (which you probably
==93131== don't want to do, unless you know exactly what you're doing,
==93131== or are doing some strange experiment):
==93131==   /usr/lib/x86_64-linux-gnu/valgrind/../../bin/vgdb --pid=93131 ...command...
==93131== 
==93131== TO DEBUG THIS PROCESS USING GDB: start GDB like this
==93131==   /path/to/gdb ./build/bin/dxc
==93131== and then give GDB the following command
==93131==   target remote | /usr/lib/x86_64-linux-gnu/valgrind/../../bin/vgdb --pid=93131
==93131== --pid is optional if only one valgrind process is running
==93131== 
--93131-- REDIR: 0x4022e10 (ld-linux-x86-64.so.2:strlen) redirected to 0x580c9ce2 (???)
--93131-- REDIR: 0x4022be0 (ld-linux-x86-64.so.2:index) redirected to 0x580c9cfc (???)
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_core-amd64-linux.so
--93131--    object doesn't have a symbol table
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so
--93131--    object doesn't have a symbol table
==93131== WARNING: new redirection conflicts with existing -- ignoring it
--93131--     old: 0x04022e10 (strlen              ) R-> (0000.0) 0x580c9ce2 ???
--93131--     new: 0x04022e10 (strlen              ) R-> (2007.0) 0x0483f060 strlen
--93131-- REDIR: 0x401f5f0 (ld-linux-x86-64.so.2:strcmp) redirected to 0x483ffd0 (strcmp)
--93131-- REDIR: 0x4023370 (ld-linux-x86-64.so.2:mempcpy) redirected to 0x4843a20 (mempcpy)
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libdl-2.31.so
--93131--   Considering /usr/lib/x86_64-linux-gnu/libdl-2.31.so ..
--93131--   .. CRC mismatch (computed 2bb25318 wanted f334bb32)
--93131--   Considering /lib/x86_64-linux-gnu/libdl-2.31.so ..
--93131--   .. CRC mismatch (computed 2bb25318 wanted f334bb32)
--93131--   Considering /usr/lib/debug/lib/x86_64-linux-gnu/libdl-2.31.so ..
--93131--   .. CRC is valid
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libtinfo.so.6.2
--93131--    object doesn't have a symbol table
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
--93131--   Considering /usr/lib/debug/.build-id/f0/983025f0e0f327a6da752ff4ffa675e0be393f.debug ..
--93131--   .. build-id is valid
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28
--93131--    object doesn't have a symbol table
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libm-2.31.so
--93131--   Considering /usr/lib/x86_64-linux-gnu/libm-2.31.so ..
--93131--   .. CRC mismatch (computed 3cfca93f wanted 658b15c9)
--93131--   Considering /lib/x86_64-linux-gnu/libm-2.31.so ..
--93131--   .. CRC mismatch (computed 3cfca93f wanted 658b15c9)
--93131--   Considering /usr/lib/debug/lib/x86_64-linux-gnu/libm-2.31.so ..
--93131--   .. CRC is valid
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libgcc_s.so.1
--93131--    object doesn't have a symbol table
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libc-2.31.so
--93131--   Considering /usr/lib/x86_64-linux-gnu/libc-2.31.so ..
--93131--   .. CRC mismatch (computed ef41b1a0 wanted f854b801)
--93131--   Considering /lib/x86_64-linux-gnu/libc-2.31.so ..
--93131--   .. CRC mismatch (computed ef41b1a0 wanted f854b801)
--93131--   Considering /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.31.so ..
--93131--   .. CRC is valid
--93131-- REDIR: 0x4caa4b0 (libc.so.6:memmove) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca97b0 (libc.so.6:strncpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa7e0 (libc.so.6:strcasecmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca90d0 (libc.so.6:strcat) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9810 (libc.so.6:rindex) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cabc80 (libc.so.6:rawmemchr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc6d10 (libc.so.6:wmemchr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc6850 (libc.so.6:wcscmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa610 (libc.so.6:mempcpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa440 (libc.so.6:bcmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9740 (libc.so.6:strncmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9180 (libc.so.6:strcmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa570 (libc.so.6:memset) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc6810 (libc.so.6:wcschr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca96a0 (libc.so.6:strnlen) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9260 (libc.so.6:strcspn) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa830 (libc.so.6:strncasecmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9200 (libc.so.6:strcpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa980 (libc.so.6:memcpy@@GLIBC_2.14) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc7f80 (libc.so.6:wcsnlen) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc6890 (libc.so.6:wcscpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9850 (libc.so.6:strpbrk) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9130 (libc.so.6:index) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9660 (libc.so.6:strlen) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cb2bd0 (libc.so.6:memrchr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa880 (libc.so.6:strcasecmp_l) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa400 (libc.so.6:memchr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cc6960 (libc.so.6:wcslen) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca9b10 (libc.so.6:strspn) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa780 (libc.so.6:stpncpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa720 (libc.so.6:stpcpy) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4cabcc0 (libc.so.6:strchrnul) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa8d0 (libc.so.6:strncasecmp_l) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4ca96e0 (libc.so.6:strncat) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4d37880 (libc.so.6:__memcpy_chk) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4caa320 (libc.so.6:strstr) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4d92410 (libc.so.6:__strrchr_avx2) redirected to 0x483ea10 (rindex)
--93131-- REDIR: 0x4ca4110 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
--93131-- REDIR: 0x4d925e0 (libc.so.6:__strlen_avx2) redirected to 0x483ef40 (strlen)
--93131-- REDIR: 0x4d8ebd0 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x48421e0 (bcmp)
--93131-- REDIR: 0x4d8dae0 (libc.so.6:__strcmp_avx2) redirected to 0x483fed0 (strcmp)
--93131-- REDIR: 0x4966b20 (libstdc++.so.6:operator new(unsigned long)) redirected to 0x483bdf0 (operator new(unsigned long))
--93131-- REDIR: 0x4966b60 (libstdc++.so.6:operator new(unsigned long, std::nothrow_t const&)) redirected to 0x483c080 (operator new(unsigned long, std::nothrow_t const&))
--93131-- REDIR: 0x4d955f0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x48429f0 (memmove)
--93131-- REDIR: 0x4966b80 (libstdc++.so.6:operator new[](unsigned long)) redirected to 0x483c510 (operator new[](unsigned long))
--93131-- REDIR: 0x4d97110 (libc.so.6:__wcsnlen_avx2) redirected to 0x4844080 (wcsnlen)
--93131-- REDIR: 0x4ca5b40 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
--93131-- REDIR: 0x4ca4700 (libc.so.6:free) redirected to 0x483c9d0 (free)
--93131-- Reading syms from /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/libtinfo.so.5.9
--93131--    object doesn't have a symbol table
--93131-- Reading syms from /usr/lib/x86_64-linux-gnu/librt-2.31.so
--93131--   Considering /usr/lib/x86_64-linux-gnu/librt-2.31.so ..
--93131--   .. CRC mismatch (computed a25149bb wanted cf42d776)
--93131--   Considering /lib/x86_64-linux-gnu/librt-2.31.so ..
--93131--   .. CRC mismatch (computed a25149bb wanted cf42d776)
--93131--   Considering /usr/lib/debug/lib/x86_64-linux-gnu/librt-2.31.so ..
--93131--   .. CRC is valid
--93131-- REDIR: 0x4964d70 (libstdc++.so.6:operator delete(void*)) redirected to 0x483cf50 (operator delete(void*))
--93131-- REDIR: 0x4d95a70 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x48428e0 (memset)
--93131-- REDIR: 0x4d955e0 (libc.so.6:__memcpy_chk_avx_unaligned_erms) redirected to 0x4843b10 (__memcpy_chk)
--93131-- REDIR: 0x4d8df20 (libc.so.6:__strncmp_avx2) redirected to 0x483f670 (strncmp)
--93131-- REDIR: 0x4d91ff0 (libc.so.6:__strchr_avx2) redirected to 0x483ebf0 (index)
--93131-- REDIR: 0x4d8e440 (libc.so.6:__memchr_avx2) redirected to 0x4840050 (memchr)
--93131-- REDIR: 0x4d92220 (libc.so.6:__strchrnul_avx2) redirected to 0x4843540 (strchrnul)
--93131-- REDIR: 0x4d93eb0 (libc.so.6:__strncpy_avx2) redirected to 0x483f230 (strncpy)
--93131-- REDIR: 0x4d92780 (libc.so.6:__strnlen_avx2) redirected to 0x483eee0 (strnlen)
--93131-- REDIR: 0x4d96f50 (libc.so.6:__wcslen_avx2) redirected to 0x4844050 (wcslen)
--93131-- REDIR: 0x4964da0 (libstdc++.so.6:operator delete[](void*)) redirected to 0x483d6e0 (operator delete[](void*))
--93131-- REDIR: 0x4d961c0 (libc.so.6:__wcscmp_avx2) redirected to 0x48440e0 (wcscmp)
--93131-- REDIR: 0x4cc6a00 (libc.so.6:wcsncmp) redirected to 0x48331d0 (_vgnU_ifunc_wrapper)
--93131-- REDIR: 0x4d96610 (libc.so.6:__wcsncmp_avx2) redirected to 0x4844120 (wcsncmp)
--93131-- REDIR: 0x4964d80 (libstdc++.so.6:operator delete(void*, unsigned long)) redirected to 0x483d160 (operator delete(void*, unsigned long))
--93131-- memcheck GC: 1000 nodes, 655 survivors (65.5%)
--93131-- memcheck GC: 1414 new table size (stepup)
--93131-- memcheck GC: 1414 nodes, 1103 survivors (78.0%)
--93131-- memcheck GC: 1999 new table size (stepup)
--93131-- memcheck GC: 1999 nodes, 1733 survivors (86.7%)
--93131-- memcheck GC: 2827 new table size (stepup)
--93131-- REDIR: 0x4964d90 (libstdc++.so.6:operator delete(void*, std::nothrow_t const&)) redirected to 0x483d370 (operator delete(void*, std::nothrow_t const&))
--93131-- memcheck GC: 2827 nodes, 2254 survivors (79.7%)
--93131-- memcheck GC: 3997 new table size (stepup)
--93131-- memcheck GC: 3997 nodes, 2851 survivors (71.3%)
--93131-- memcheck GC: 5652 new table size (stepup)
==93131== Invalid read of size 8
==93131==    at 0x53EF3A0: llvm::BasicBlock::getTerminator() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58B024F: llvm::SimplifyCFG(llvm::BasicBlock*, llvm::TargetTransformInfo const&, unsigned int, llvm::AssumptionCache*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58057DA: simplifyFunctionCFG(llvm::Function&, llvm::TargetTransformInfo const&, llvm::AssumptionCache*, int) [clone .constprop.102] (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4156: llvm::FPPassManager::runOnFunction(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F499D: llvm::legacy::FunctionPassManagerImpl::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4AF5: llvm::legacy::FunctionPassManager::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58EC9B0: clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::raw_pwrite_stream*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D3035: clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x61A8754: clang::ParseAST(clang::Sema&, bool, bool) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D6FA0: clang::CodeGenAction::ExecuteAction() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x5A79455: clang::FrontendAction::Execute() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x539D269: DxcCompiler::Compile(DxcBuffer const*, wchar_t const**, unsigned int, IDxcIncludeHandler*, GUID const&, void**) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==  Address 0x38 is not stack'd, malloc'd or (recently) free'd
==93131== 
==93131== 
==93131== Process terminating with default action of signal 11 (SIGSEGV)
==93131==  Access not within mapped region at address 0x38
==93131==    at 0x53EF3A0: llvm::BasicBlock::getTerminator() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58B024F: llvm::SimplifyCFG(llvm::BasicBlock*, llvm::TargetTransformInfo const&, unsigned int, llvm::AssumptionCache*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58057DA: simplifyFunctionCFG(llvm::Function&, llvm::TargetTransformInfo const&, llvm::AssumptionCache*, int) [clone .constprop.102] (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4156: llvm::FPPassManager::runOnFunction(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F499D: llvm::legacy::FunctionPassManagerImpl::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4AF5: llvm::legacy::FunctionPassManager::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58EC9B0: clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::raw_pwrite_stream*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D3035: clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x61A8754: clang::ParseAST(clang::Sema&, bool, bool) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D6FA0: clang::CodeGenAction::ExecuteAction() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x5A79455: clang::FrontendAction::Execute() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x539D269: DxcCompiler::Compile(DxcBuffer const*, wchar_t const**, unsigned int, IDxcIncludeHandler*, GUID const&, void**) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==  If you believe this happened as a result of a stack
==93131==  overflow in your program's main thread (unlikely but
==93131==  possible), you can try to increase the size of the
==93131==  main thread stack using the --main-stacksize= flag.
==93131==  The main thread stack size used in this run was 8388608.
==93131== 
==93131== HEAP SUMMARY:
==93131==     in use at exit: 1,178,696 bytes in 2,741 blocks
==93131==   total heap usage: 3,587 allocs, 846 frees, 1,961,613 bytes allocated
==93131== 
==93131== Searching for pointers to 2,741 not-freed blocks
==93131== Checked 1,072,368 bytes
==93131== 
==93131== LEAK SUMMARY:
==93131==    definitely lost: 0 bytes in 0 blocks
==93131==    indirectly lost: 0 bytes in 0 blocks
==93131==      possibly lost: 42,060 bytes in 696 blocks
==93131==    still reachable: 1,136,636 bytes in 2,045 blocks
==93131==                       of which reachable via heuristic:
==93131==                         multipleinheritance: 496 bytes in 1 blocks
==93131==         suppressed: 0 bytes in 0 blocks
==93131== Rerun with --leak-check=full to see details of leaked memory
==93131== 
==93131== ERROR SUMMARY: 2 errors from 1 contexts (suppressed: 0 from 0)
==93131== 
==93131== 2 errors in context 1 of 1:
==93131== Invalid read of size 8
==93131==    at 0x53EF3A0: llvm::BasicBlock::getTerminator() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58B024F: llvm::SimplifyCFG(llvm::BasicBlock*, llvm::TargetTransformInfo const&, unsigned int, llvm::AssumptionCache*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58057DA: simplifyFunctionCFG(llvm::Function&, llvm::TargetTransformInfo const&, llvm::AssumptionCache*, int) [clone .constprop.102] (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4156: llvm::FPPassManager::runOnFunction(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F499D: llvm::legacy::FunctionPassManagerImpl::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x63F4AF5: llvm::legacy::FunctionPassManager::run(llvm::Function&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58EC9B0: clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::raw_pwrite_stream*) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D3035: clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x61A8754: clang::ParseAST(clang::Sema&, bool, bool) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x58D6FA0: clang::CodeGenAction::ExecuteAction() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x5A79455: clang::FrontendAction::Execute() (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==    by 0x539D269: DxcCompiler::Compile(DxcBuffer const*, wchar_t const**, unsigned int, IDxcIncludeHandler*, GUID const&, void**) (in /home/jack/code/vulkan/1.3.204.1/x86_64/lib/libdxcompiler.so.3.7)
==93131==  Address 0x38 is not stack'd, malloc'd or (recently) free'd
==93131== 
==93131== ERROR SUMMARY: 2 errors from 1 contexts (suppressed: 0 from 0)
Segmentation fault (core dumped)
```

**Comment by pow2clk** [comment]
Hi @Jack-Clark!

Thanks for the report. It looks like #4446 fixed this on the same day you reported it! Wish we could do that more often!

Closing this bug, please let us know if you have the problem with the compiler after that change.

**Comment by Jack-Clark** [comment]
Hi @pow2clk, I've rebuilt dxc from main using commit id (76fed39c368cf65167e6db00738b212dac024b49) and I still get segfault errors that look like they have the same root cause (the valgrind output is almost exactly identical). From a file of about 23k tests created by a fuzzer, I get 588 segfaults. I've attached an archive containing all the hlsl tests that caused a segfault in this test run.
[hlsl_segfaults.tar.gz](https://github.com/microsoft/DirectXShaderCompiler/files/9038888/hlsl_segfaults.tar.gz)
